### PR TITLE
[PD] keep hole size when changing type

### PR DESCRIPTION
**Pull request by donovaly**
as mentioned in https://forum.freecadweb.org/viewtopic.php?f=13&t=47584&start=10#p433116
a typical use case is that one changes the hole type from e.g. ISO normal to ISO fine. The size should however be the same.

This PR enables this, see the screencast:

4rnlmbduB4

### CODE REVIEW

File: src/Mod/PartDesign/Gui/TaskHoleParameters.cpp [hunk 1]
```diff
@@ -285,8 +285,59 @@ void TaskHoleParameters::threadTypeChanged(int index)
         return;
 
     PartDesign::Hole* pcHole = static_cast<PartDesign::Hole*>(vp->getObject());
+
+    // A typical case is taht users change from an ISO profile to another one.
```
Comment (tobymurray) [inline src/Mod/PartDesign/Gui/TaskHoleParameters.cpp:289]: Minor, spelling: `taht` => `that`

File: src/Mod/PartDesign/Gui/TaskHoleParameters.cpp [hunk 2]
```diff
@@ -285,8 +285,59 @@ void TaskHoleParameters::threadTypeChanged(int index)
         return;
 
     PartDesign::Hole* pcHole = static_cast<PartDesign::Hole*>(vp->getObject());
+
+    // A typical case is taht users change from an ISO profile to another one.
+    // When they had e.g. the size "M3" in one profile they expect
+    // the same size in the other profile if it exists there.
+    // To achive this, we check the size combobox:
```
Comment (tobymurray) [inline src/Mod/PartDesign/Gui/TaskHoleParameters.cpp:292]: Minor, spelling: `achive` => `achieve `

**Comment by donovaly** [comment]
Thanks for the review. I fixed the typos in the comment.

I also extended the patch to take care of the other properties that are currently reset when changing the type.

**Comment by wwmayer** [comment]
Merged.

See 6e6f2c8a0a for some improvements.

**Comment by donovaly** [comment]
> See [6e6f2c8](https://github.com/FreeCAD/FreeCAD/commit/6e6f2c8a0a1be3a3a2449ea3b7b6ee4465bd5cce) for some improvements.

Thank you.
(I must admit I never read the addItem()
 https://doc.qt.io/qt-5/qcombobox.html#addItem
and thus was not aware of the user data as second argument.)

**Pull request closed without merging**